### PR TITLE
EWS: alerts change

### PR DESCRIPTION
**Pull request by kkrampa**
@gcapalbo Could you take a look on this?
Probably easiest to read commit by commit

### CODE REVIEW

**Comment by gcapalbo** [comment]
@kkrampa looks like a few things are still outstanding:

- see my comment about `TEST_BACKEND`
- I see you replaced `clean_phone_number` with `strip_plus`, but we still want to clean the phone number. I think it would be best if we just used `apply_leniency` instead
- can you confirm what the datatype is for the phone numbers when they come from the source?
- can you confirm that when we set the `.phone_numbers` property for mobile workers that we are setting the phone number after `apply_leniency` has been called?

**Comment by kkrampa** [comment]
@gcapalbo 
```prepare_domain``` is only used in tests. Nonetheless I changed backend.
Phone numbers are always coming as strings from EWS api. 
Now ```apply_leniency``` is aplied before phone number is assigned to user.

**Comment by gcapalbo** [comment]
@kkrampa Sorry, I didn't realize that `prepare_domain` was only used in tests.  But I like what you have added now.  Just one thing - we should move the creation of the `TestSMSBackend` and its `BackendMapping` to the `setUp` method of the test case, and make sure that we delete them on the `tearDown` method.

**Comment by kkrampa** [comment]
Updated but it seems build on travis doesn't work

**Comment by czue** [comment]
they're very slow

File: custom/ewsghana/utils.py [hunk 1]
```diff
@@ -1,23 +1,28 @@
 from decimal import Decimal
+from couchdbkit import ResourceNotFound
```
Comment (dimagimon) [inline custom/ewsghana/utils.py]: F401 'ResourceNotFound' imported but unused

File: custom/ewsghana/utils.py [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 from corehq.apps.accounting import generator
 from corehq.apps.accounting.models import BillingAccount, DefaultProductPlan, SoftwarePlanEdition, Subscription
 from corehq.apps.commtrack.models import StockState, SupplyPointCase
+from corehq.apps.domain.shortcuts import create_domain
 from corehq.apps.locations.models import SQLLocation, LocationType, Location
 from datetime import timedelta, datetime
 from dateutil import rrule
 from dateutil.rrule import MO
 from django.utils import html
+from corehq.apps.sms.mixin import VerifiedNumber, BackendMapping, MobileBackend
```
Comment (dimagimon) [inline custom/ewsghana/utils.py]: F401 'MobileBackend' imported but unused

**Comment by gcapalbo** [comment]
@kkrampa Looks good, I'm just waiting on the build before merging.  I cancelled all builds except the one on the latest commit to hopefully speed things up.

**Comment by gcapalbo** [comment]
@kkrampa It looks like the build process is still behind - are you able to run all of these tests locally and have them pass? If so, then I'll merge

**Comment by kkrampa** [comment]
I've just run all tests for EWS module and they passed.

**Comment by gcapalbo** [comment]
@kkrampa Ok, I just merged. One thing I forgot to mention was that all sms sent to web users will always use the domain's default backend when sending the sms.  Can you confirm that the domain's default backend is currently set to the test backend for all of the projects that use this functionality?

**Comment by kkrampa** [comment]
@gcapalbo Yes, TEST BACKEND is set for all existing EWS domains..

**Comment by czue** [comment]
@kkrampa possible this introduced this test failure?

```
======================================================================
ERROR: setUpClass (custom.ewsghana.tests.ews_test_reminders.EWSTestReminders)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/testrunner.py", line 138, in new_setUpClass
    result = original_setUpClass(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/testrunner.py", line 138, in new_setUpClass
    result = original_setUpClass(*args, **kwargs)
  File "/home/travis/build/dimagi/commcare-hq/custom/ewsghana/tests/ews_test_reminders.py", line 17, in setUpClass
    cls.sms_backend_mapping, cls.backend = create_backend()
  File "/home/travis/build/dimagi/commcare-hq/custom/ewsghana/utils.py", line 140, in create_backend
    backend.save()
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/couchdbkit/schema/base.py", line 134, in save
    db.save_doc(doc, **params)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/couchdbkit/client.py", line 515, in save_doc
    **params).json_body
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/restkit/resource.py", line 153, in put
    headers=headers, params_dict=params_dict, **params)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/couchdbkit/resource.py", line 140, in request
    response=e.response)
ResourceConflict: Document update conflict.
```

**Comment by czue** [comment]
@kkrampa @gcapalbo reverted this to avoid many more possibly broken builds. @kkrampa feel free to reopen but lets wait for the real build this time

**Comment by gcapalbo** [comment]
@kkrampa It looks like the problem is in: https://github.com/lwyszomi/commcare-hq/blob/50bd35c2adfea4caf0046004d047a53a1f9bdec1/custom/ilsgateway/tests/test_migration_task.py

Just need to add a tearDown method to that test which deletes the sms backend created there

**Pull request merged**